arduino-ide-extension/src/browser/settings.tsx

```diff
@@ -376,9 +378,18 @@ export class SettingsComponent extends React.Component<SettingsComponent.Props,
                     onChange={this.additionalUrlsDidChange} />
                 <i className='fa fa-window-restore theia-button shrink' onClick={this.editAdditionalUrlDidClick} />
             </div>
+
+            <div className='flex-line'>
+                <button className='theia-button shrink' onClick={this.openExtendedSettings}>Extended Settings</button>
+            </div>
```

Comment:
@kittaakos This is just a draft button in the Arduino settings panel. When clicked it opens the "theia" settings in a tab, and - supposedly - closes the current Arduino settings panel.

I don't really like the way I close this dialog, as I don't think I need an extra command, as the `SettingsComponent` shouldn't be aware it's called inside a dialog.

In my opinion a better approach should be sending a message to the parent (Widget) stating that the extended settings are being opened. The Widget should communicate somehow to the Dialog to call the `.close()`.

Reply:
> as I don't think I need an extra command

I agree. Let's do it without a command.